packages/rsocket-core/src/RSocketConnector.ts

```diff
@@ -47,22 +50,36 @@ export class RSocketConnector {
 
   async connect(): Promise<RSocket> {
     const connection = await this.transport.connect();
-
-    const multiplexer = new ClientServerInputMultiplexerDemultiplexer(
+    const keepAliveSender = new KeepAliveSender(
+      connection.multiplexer.connectionOutbound,
+      this.setupFrame.keepAlive
+    );
+    const keepAliveHandler = new KeepAliveHandler(
+      connection,
+      this.setupFrame.lifetime
+    );
+    const connectionFrameHandler = new ConnectionFrameHandler(
       connection,
-      {
-        fragmentSize: 0,
-        keepAlive: {
-          timeout: this.setupFrame.lifetime,
-          period: this.setupFrame.keepAlive,
-        },
-        streamIdSupplier: () => 1,
-        responder: this.responder,
-      }
+      keepAliveHandler,
+      this.responder
     );
+    const streamsHandler = new StreamHandler(this.responder, 0);
 
-    connection.send(this.setupFrame);
+    connection.onClose((e) => {
+      keepAliveSender.close();
+      keepAliveHandler.close();
+      connectionFrameHandler.close(e);
+    });
+    connection.demultiplexer.handleConnectionFrames(
+      connectionFrameHandler.handle.bind(connectionFrameHandler)
+    );
+    connection.demultiplexer.handleStream(
+      streamsHandler.handle.bind(connectionFrameHandler)
+    );
+    connection.multiplexer.connectionOutbound.send(this.setupFrame);
```

Comment:
Debating if `ClientServerInputMultiplexerDemultiplexer` shouldn't be abstract and instead constructed and injected where needed? Does `ClientServerInputMultiplexerDemultiplexer` being abstract provide us with more flexibility for later transport implementations?

Reply:
It can be non-abstract and can be constructed where it is needed. But I don't see much value in doing so

Reply:
ok. I'll think on this more but likely won't worry about it until later.